### PR TITLE
JNDI additions and ability to override the durable subscription prefix

### PR DESCRIPTION
**Pull request by AndySavin**
I've re-forked and started again.  Changes are as follows:

Updated JNDI factory to allow specification of SNS and SQS endpoints, the SQSConnectorFactory class name, the max poll wait time and an override for the durable subscription prefix.

I've also updated the NevadoSession to use this prefix instead of the default if it is set.  Amazon seems to have an 80 character limit on the name of a subscriber and nevado_durable_topic took up a bit too much of that allocation for our needs.

I don't think I've changed anything that would require a unit test however, let me know and I'll try and create some.

Andy

### CODE REVIEW

File: nevado-jms/src/main/java/org/skyscreamer/nevado/jms/NevadoSession.java [hunk 1]
```diff
@@ -344,6 +344,10 @@ public void unsubscribe(String name) throws JMSException
 
     protected String getDurableEndpointQueueName(String durableSubscriptionName) {
         String queueName = NevadoProviderQueuePrefix.DURABLE_SUBSCRIPTION_PREFIX + durableSubscriptionName;
+        if (_connection.getDurableSubcriptionPrefixOveride() != null)
```
Comment (carterpage) [inline nevado-jms/src/main/java/org/skyscreamer/nevado/jms/NevadoSession.java:347]: I'd prefer an if/else or a = b ? c : d so we only set queueName once (for code clarity).

File: nevado-jms/src/main/java/org/skyscreamer/nevado/jms/resource/NevadoReferencableFactory.java [hunk 2]
```diff
@@ -47,6 +49,32 @@ public Object getObjectInstance(Object obj, Name name, Context ctx, Hashtable en
                 {
                     connectionFactory.setOverrideJMSTTL(Long.parseLong(jmsTtl));
                 }
+                String sqsEndpoint = getRefContent(ref, NevadoConnectionFactory.JNDI_SQS_ENDPOINT);
```
Comment (carterpage) [inline nevado-jms/src/main/java/org/skyscreamer/nevado/jms/resource/NevadoReferencableFactory.java:52]: Please update https://github.com/skyscreamer/nevado/blob/master/nevado-jms/src/test/java/org/skyscreamer/nevado/jms/resource/ReferencableTest.java to include tests for these five fields.

**Comment by carterpage** [comment]
Comments added.

**Comment by AndySavin** [comment]
Hi, I've made those changes.  I had looked to see if there was a test already clearly I missed it.
Andy

**Comment by carterpage** [comment]
Looks good!  I'll try to cut a new release this weekend.  Thanks for the contribution.

**Pull request merged**